### PR TITLE
feat: add date filters for activity completion history

### PR DESCRIPTION
**Pull request by gnapse**
## Summary
- expose inclusive `dateFrom` and exclusive `dateTo` filters on `find-activity`
- guide completion-history queries, including recurring task occurrences, to activity events
- add date-range forwarding coverage

### CODE REVIEW

File: src/tools/find-activity.ts [hunk 1]
```diff
@@ -38,6 +38,20 @@ const ArgsSchema = {
 
     initiatorId: z.string().optional().describe('Filter by the user ID who initiated the event.'),
 
+    dateFrom: z
+        .string()
```
Comment (doistbot) [inline src/tools/find-activity.ts]: <a href="#">P2</a>  These fields accept any string despite documenting an ISO 8601 date/date-time contract, so values such as `"tomorrow"` or `"not-a-date"` are forwarded directly to `getActivityLogs`. Validate both `dateFrom` and `dateTo` as ISO dates or date-times (and add rejection coverage) so malformed filter input fails at the tool boundary.

**Review by doistbot — COMMENTED** [review]
This PR adds inclusive `dateFrom` and exclusive `dateTo` filters to `find-activity`, forwarding them to the Todoist API and surfacing them in the tool output.

Few things worth tightening:

- `dateFrom` and `dateTo` accept arbitrary strings despite documenting an ISO 8601 contract — consider validating them as dates or date-times (and adding rejection tests) so malformed values like `"tomorrow"` fail at the tool boundary rather than being passed through to the API.

I also left one optional follow-up note in the details below.

<details>
<summary>Optional follow-up note (1)</summary>

- **<a href="#">P3</a> src/tools/find-activity.ts:41**: Use the existing `optionalString` helper for both new date fields. It normalizes LLM-supplied empty strings to `undefined`; with the current schemas, an empty date is omitted from the API request but remains in `appliedFilters`, incorrectly implying that filter was applied.

</details>

<sub>[Share Feedback](https://github.com/Doist/doistbot-app/issues/new?template=review-feedback.yml&title=Review+feedback+for+Doist%2Ftodoist-mcp&pr_url=https%3A%2F%2Fgithub.com%2FDoist%2Ftodoist-mcp%2Fpull%2F553&datadog_url=https%3A%2F%2Fapp.datadoghq.com%2Flogs%3Fquery%3Dservice%253A%2528doistbot-app-sandbox%2520OR%2520doistbot-app%2529%2520%2540meta.deliveryId%253Ad2ee6160-8f8b-11f1-85f6-0b05afcb824e%26agg_m%3Dcount%26agg_m_source%3Dbase%26agg_t%3Dcount%26clustering_pattern_field_path%3Dmessage%26cols%3Dhost%252Cservice%26fromUser%3Dtrue%26messageDisplay%3Dinline%26refresh_mode%3Dsliding%26storage%3Dhot%26stream_sort%3Ddesc%26viz%3Dstream%26live%3Dtrue) • [Review Logs](https://app.datadoghq.com/logs?query=service%3A%28doistbot-app-sandbox%20OR%20doistbot-app%29%20%40meta.deliveryId%3Ad2ee6160-8f8b-11f1-85f6-0b05afcb824e&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&live=true)</sub>

File: src/tools/find-activity.ts (hunk 1 above)
Comment (gnapse) [reply src/tools/find-activity.ts]: Addressed in 6b8274d: both fields now accept only ISO `YYYY-MM-DD` dates or ISO date-times with a timezone offset; natural-language values such as `tomorrow` are rejected at MCP input validation. I also used `optionalString` so empty values normalize to `undefined` and do not appear as applied filters.
